resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/filesystem/FileStoreCreationRule.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *******************************************************************************/
+package org.eclipse.core.tests.filesystem;
+
+import java.net.URI;
+import org.eclipse.core.filesystem.EFS;
+import org.eclipse.core.filesystem.IFileStore;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.tests.harness.FileSystemHelper;
+import org.eclipse.core.tests.internal.filesystem.ram.MemoryTree;
+import org.junit.rules.ExternalResource;
+
+/**
+ * A test rule for automatically creating and disposing a file store for the
+ * local file system or in memory.
+ */
+public class FileStoreCreationRule extends ExternalResource {
+	public enum FileSystemType {
+		LOCAL, IN_MEMORY
+	}
+
+	private final FileSystemType fileSystemType;
+
+	private IFileStore fileStore;
+
+	public FileStoreCreationRule(FileSystemType fileSystemType) {
+		this.fileSystemType = fileSystemType;
+	}
+
+	public IFileStore getFileStore() {
+		return fileStore;
+	}
+
+	@Override
+	protected void before() throws Throwable {
+		switch(fileSystemType) {
+		case LOCAL:
+			var fileStoreLocation = FileSystemHelper.getRandomLocation(FileSystemHelper.getTempDir());
```

Comment:
Would it be possible to attach the name of the requestor (the test method and/or class) to the path so that it makes it easier to find the culprit when some resources are not cleaned up properly?
This could even go into the method `getRandomLocation` itself so that even other tests still not using the `FileStoreCreationRule` can benefit from it.

It doesn't need to happen in this PR, just something to keep in mind.

Reply:
That's a good idea. I modified the location calculation to append the test name in dec3328.

But you are right that this makes sense for every test, so we should probably enhance the `FileSystemHelper` to accept the test name (it cannot identify it on its own) and add it to generated locations.